src/DataLib/fendl32B_retrofit/groupr_tools.py

```diff
@@ -91,9 +91,10 @@ def fill_input_template(material_id, MTs, element, A, mt_dict):
 
     card9_lines = []
     for MT in MTs:
```

Comment:
I think you can use set algebra here to find the intersection of `MTs` and `mt_dict.keys()` and avoid the if statement.
```suggestion
    for MT in {MTs} & mt_dict.keys():
```